src/cli/format.rs

```diff
@@ -70,15 +70,16 @@ pub fn format_execute_with_args(args: FormatArgs, global_options: GlobalOptions)
 
     let socket = DaemonSocketPath::from_socket_dir(&socket_dir);
 
-    if !daemon_is_alive(&socket)? {
+    let daemon_status = daemon_is_alive(&socket)?;
+    if matches!(daemon_status, DaemonStatus::NotRunning) {
         start_daemon(&socket, false)?;
     }
```

Comment:
Originally, the daemon restart should have been performed here rather than inside `run_command`.

- Move the restart process from `run_command` to `format_execute_with_args`, etc.
  - Likely the process of “checking if the daemon is alive, starting it if it’s not, and restarting if the build ID is different” is common to `format`, `bulk-format`, and similar commands, so this logic should be extracted into its own function.
- Continue to let `run_command` accept a boolean `check_alive` parameter, which indicates whether or not to check if the daemon is alive.

I apologize that these instructions differ from the previous ones, but thank you for your understanding and cooperation.